migrator/clone/postgres/db_clone_manager_impl.go

```diff
@@ -162,24 +162,26 @@ func (d *dbCloneManagerImpl) GetCloneToMigrate(rocksVersion *migrations.Migratio
 		// Rocks more recently updated than Postgres so need to migrate from there.  Otherwise, Postgres is more recent
 		// so just fall through to the rest of the processing.
 		if currClone.GetMigVersion().LastPersisted.Before(rocksVersion.LastPersisted) {
-			// Rollback is not enabled, so process the Rocks -> Postgres migration on CurrentClone.
-			if !d.rollbackEnabled() {
-				return CurrentClone, true, nil
-			}
-
-			// Create a temp clone for processing of current
-			// Seed it from an empty database because we need to run migrations from Rocks to Postgres.
-			if !d.databaseExists(TempClone) {
-				err := pgadmin.CreateDB(d.sourceMap, d.adminConfig, pgadmin.EmptyDB, TempClone)
+			// We want to start fresh as we are migrating from Rocks->Postgres.  So any data that exists in
+			// Postgres from a previous upgrade followed by a rollback needs to be ignored.  So just drop current
+			// and let it create anew.
+			err := pgadmin.DropDB(d.sourceMap, d.adminConfig, CurrentClone)
+			if err != nil {
+				log.Errorf("Unable to drop current clone: %v", err)
 
-				// If for some reason, we cannot create a temp clone we will need to continue to upgrade
-				// with the current and thus no fallback.
+				// The drop of the current clone is unlikely to fail, in the event it does; we can try to
```

Comment:
What could be potential reason for not being able to Drop database? I cannot think of the case that could be corrected by using temp.

Reply:
yeah that is probably over programming on my part.  the only ways that could fail generally are because we are in a very bad state.  It could fail if we are trying to drop the database from the same connection we are using to drop it.  But we aren't because we are using the admin config to do this.  I will clean some of this stuff up.